backend/api/Areas/Admin/Controllers/ParcelController.cs

```diff
@@ -183,73 +197,118 @@ public IActionResult AddParcels([FromBody] ParcelModel[] models)
+                        _mapper.Map (building, b_entity);
                         b_entity.UpdatedById = userId;
                         b_entity.UpdatedOn = DateTime.UtcNow;
-                        _dbContext.Buildings.Update(b_entity);
+
+                        // Update evaluation.
+                        var b_eval = b_entity.Evaluations.FirstOrDefault (e => e.FiscalYear == building.FiscalYear);
+
+                        if (b_eval == null)
+                        {
+                            b_entity.Evaluations.Add (new Entity.BuildingEvaluation (building.FiscalYear, b_entity) // TODO: Move this logic to AutoMapper.
+                                {
+                                    EstimatedValue = building.EstimatedValue,
+                                        AssessedValue = building.AssessedValue,
+                                        NetBookValue = building.NetBookValue,
+                                        CreatedById = userId
+                                });
+                        }
+                        else
+                        {
+                            b_eval.EstimatedValue = building.EstimatedValue;
+                            b_eval.AssessedValue = building.AssessedValue;
+                            b_eval.NetBookValue = building.NetBookValue;
+                            b_eval.UpdatedById = userId; // TODO: Move to DAL.
+                            b_eval.UpdatedOn = DateTime.UtcNow;
+                        }
+
+                        _pimsAdminService.Building.UpdateOne (b_entity);
 
                         // Check if the address was updated.
-                        if (b_address != b_entity.Address.ToString())
+                        if (b_address != b_entity.Address.ToString ())
                         {
                             b_entity.Address.UpdatedById = userId;
                             b_entity.Address.UpdatedOn = DateTime.UtcNow;
-                            _dbContext.Addresses.Update(b_entity.Address);
+                            _pimsAdminService.Address.UpdateOne (b_entity.Address);
                         }
                     }
                     else
                     {
-                        _logger.LogDebug($"Invalid - Attempting to update parcel with invalid building - PID:{entity.PID}, BuildingId:{building.Id}");
+                        _logger.LogDebug ($"Invalid - Attempting to update parcel with invalid building - PID:{entity.PID}, BuildingId:{building.Id}");
                     }
                 }
             }
 
             // Check if the address was updated.
-            if (address != entity.Address.ToString())
+            if (address != entity.Address.ToString ())
             {
                 entity.Address.UpdatedById = userId;
                 entity.Address.UpdatedOn = DateTime.UtcNow;
-                _dbContext.Addresses.Update(entity.Address);
+                _pimsAdminService.Address.UpdateOne (entity.Address);
```

Comment:
Can you explain your choice of UpdateOne vs Update (which both seem to be for updating a single entity)?